client/components/course/Sidebar/Relationship.vue

```diff
@@ -74,8 +94,11 @@ export default {
       return get(activity, `refs.${this.type}`, []);
     },
     onRelationshipChanged(associations) {
+      const ref = Array.isArray(associations)
+        ? map(associations, 'id')
+        : [get(associations, 'id')];
       let activity = cloneDeep(this.activity) || {};
-      set(activity, `refs.${this.type}`, map(associations, 'id'));
+      set(activity, `refs.${this.type}`, ref);
       this.update(activity);
```

Comment:
If you have time investigate could we get rid of `deepClone` here 💡

Reply:
Why?

Reply:
Because it pulls in quite complex lodash implementation (both heavy and slow) and in most cases we know data layout in advance plus don't have deeply nested tree.